src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactory.cs

```diff
@@ -38,6 +38,11 @@ public sealed class RoslynCodeTaskFactory : ITaskFactory
             "System.IO",
             "System.Linq",
             "System.Text",
+
+            #if RUNTIME_TYPE_NETCORE || NETCOREAPP
+            "System.Text.Json",
+            "System.Memory",
```

Comment:
This adds the default namespaces, but I think you'll also need to add a reference to the "common assembly references" section below.

Reply:
What about `System.Text.Json.Nodes`? I guess I could manually add the namespace myself, just wanted to ask because I would use it :)

Reply:
I can create a new special folder where their dll can be located

Reply:
That shouldn't be necessary--you should be able to use the copies MSBuild itself will use at runtime as references (_ideally_ we'd pass a ref assembly to the compiler as that's all it needs, but I wouldn't want to increase the size of MSBuild just to support these references).